Docker works

govza opened the pull request:
Make Docker work again.

govza committed:
Use one .env file

govza committed:
Move postgres_database to postgres

govza committed:
Can use Debug in production mode

govza committed:
Avoid issues for quotted .env variables used with docker

govza committed:
Docker uses new settings.py structure

al-muammar commented on .env.example:
```diff
@@ -1,8 +1,13 @@
-POSTGRES_PASSWORD="secret123"
-POSTGRES_DB="mypgsqldb"
+# Django Settings
+ALLOWED_HOSTS = *
```
It's an environment file. You can't put whitespaces around `=`.
```sh
$ WHATEVER = 25
-bash: WHATEVER: command not found
$ WHATEVER=25
$ echo $WHATEVER
25
```

al-muammar reviewed (commented):
I would love to accept, but I can't check that it works fine. You might ask @rguliev to review. Maybe he knows docker better.

govza committed:
Prevent bash whitespace error.

rguliev commented:
Ok, I will try to help here.
@govza could you provide a "test plan"? What is a better way to test your PR?

govza committed:
Use volumes instead of bind mounts for persistent data

Have to use this, not only because it's recommended, but because of bug with docker-toolbox

govza commented:
Check Project is correctly running in docker containers:

Prerequisites:
Installed docker and docker-compose

1. Clone project to machine (if you using docker-toolbox it has to be root directory of Windows ec C:/Users/Public/dev/)
2. On windows run terminal with admin rights (docker terminal, powershell), or use bash on Linux.
3. Check your docker daemon is running 
    `docker-machine status` if it is not "Running" `docker-machine start default`
3. Go to project directory 
    `cd /path/to/quran-words`
4. Create .env file from .env.example
    `cp .env.example .env`
5. Edit .env file DJANGO_EXECUTION_ENVIRONMENT line, to run project with postgres database
    `DJANGO_EXECUTION_ENVIRONMENT=PRODUCTION`
4. If it's first time you run project on machine, create volume to store persistent data (postgres)
    `docker volume create --name=postgres`
5. Start containers
    `docker-compose up`
6. At first start it should create database for project, so you need to restart the project
    Stop running containers `Ctrl + C`
    Start them again `docker-compose up`
7. Check  containers are running:
    Django at http://127.0.0.1:8000 (for linux) or http://192.168.99.100:8000 (for windows)
8. To access running container. Start new terminal windows, and run
    `docker exec -it web01 sh`
9. Run migrations for django: 
    `python3 manage.py migrate`
10. Check database is seeded by accessing adminer on http://127.0.0.1:8080 (linux) or http://192.168.99.100:8080 (for windows)
11. Login Using: System: PostgreSQL, Server: db, User: postgres, Password: secret123, Database: mypgsqldb (information provided in .env file)
12. There should be tables filled with data quran_ayah, quran_surah, etc.

govza committed:
Documentation improved

govza committed:
Remove nginx unused

rguliev commented:
Actually I failed at 3th step:) Folder is quran-words repo, DockerWorks branch
```
$ cd quran-words
$ git branch DockerWorks
$ docker-machine start default 
Docker machine "default" does not exist. Use "docker-machine ls" to list machines. Use "docker-machine create" to add a new one.
```

al-muammar reviewed (approved):
![docker-i-see](https://user-images.githubusercontent.com/11528067/52671566-521c9600-2f13-11e9-811c-8f091325be1a.jpg)
